src/System.Management.Automation/engine/SecurityDescriptorCommandsHelperBase.cs

```diff
@@ -0,0 +1,502 @@
+        /// Gets the Path of the provided PSObject.
+        /// </summary>
+        /// <param name="instance">
+        /// The PSObject for which to obtain the path.
+        /// </param>
+        /// <returns>
+        /// The path of the provided PSObject.
+        /// </returns>
+        public static string GetPath(PSObject instance)
+        {
+            if (instance == null)
+            {
+                throw PSTraceSource.NewArgumentNullException(nameof(instance));
+            }
+            else
+            {
+                // These are guaranteed to not be null
+                // Get path
+                return instance.Properties["PSPath"].Value.ToString()!;
+            }
+        }
+
+        /// <summary>
+        /// Gets the Owner of the provided PSObject.
+        /// </summary>
+        /// <param name="instance">
+        /// The PSObject for which to obtain the Owner.
+        /// </param>
+        /// <returns>
+        /// The Owner of the provided PSObject.
+        /// </returns>
+        public static string? GetOwner(PSObject instance)
+        {
+            if (instance == null)
+            {
+                throw PSTraceSource.NewArgumentNullException(nameof(instance));
+            }
+
+            if (!(instance.BaseObject is ObjectSecurity sd))
+            {
+                throw PSTraceSource.NewArgumentNullException(nameof(instance));
+            }
+
+            // Get owner
+            try
+            {
+                IdentityReference? ir = sd.GetOwner(typeof(NTAccount));
+                return ir?.ToString();
+            }
+            catch (IdentityNotMappedException)
+            {
+                // All Acl cmdlets returning SIDs will return a string
+                // representation of the SID in all cases where the SID
+                // cannot be mapped to a proper user or group name.
+            }
+
+            // We are here since we cannot get IdentityReference from sd..
+            // So return sddl..
+            return sd.GetSecurityDescriptorSddlForm(AccessControlSections.Owner);
+        }
+
+        /// <summary>
+        /// Gets the Group of the provided PSObject.
+        /// </summary>
+        /// <param name="instance">
+        /// The PSObject for which to obtain the Group.
+        /// </param>
+        /// <returns>
+        /// The Group of the provided PSObject.
+        /// </returns>
+        public static string? GetGroup(PSObject instance)
+        {
+            if (instance == null)
+            {
+                throw PSTraceSource.NewArgumentNullException(nameof(instance));
+            }
+
+            if (!(instance.BaseObject is ObjectSecurity sd))
+            {
+                throw PSTraceSource.NewArgumentNullException(nameof(instance));
+            }
+
+            // Get Group
+            try
+            {
+                IdentityReference? ir = sd.GetGroup(typeof(NTAccount));
+                return ir?.ToString();
```

Comment:
Now it can return a null instead of null reference exception. What is original design here?